umbraco/Umbraco-CMS
# V8: Reduce number of XHR for media pickers and media folder views

### Prerequisites


### Description

_This is a V8-ify of #3874_

The media folders are rather slow to display the dropzone. When I investigated the root cause, I realized it's waiting for a lot of XHR to complete before it is displayed - a matter of figuring out which "image" types can be uploaded. 

I then stumbled upon @Shazwazza's comment [here](https://github.com/umbraco/Umbraco-CMS/blob/e3b2d427a466da7dc295e6632bf45e16851528d1/src/Umbraco.Web.UI.Client/src/common/services/mediatypehelper.service.js#L25) which is basically stating the same issue.

Currently when you open a standard media picker or a media folder, the client fetches all media types one at a time to figure out which ones are "image" types. 

The following screenshot is for a site that has the default folder, file and image type plus another folder type - a total of four media types, resulting in four XHRs when opening a media picker:

![image](https://user-images.githubusercontent.com/7405322/59145890-5b61cf00-89ea-11e9-8d65-afa371f94fcf.png)

This PR moves the logic to the server side, so the client only has to query one API method to get the allowed "image" types:

![image](https://user-images.githubusercontent.com/7405322/59145871-205f9b80-89ea-11e9-99dd-86b1abd1a4df.png)

### To test this PR

1. Create a doctype with a media picker.
2. Verify that you can select existing media with the media picker.
3. Verify that you can upload new media within the media picker.
4. Verify that you can upload new media to the media tree root.
5. Verify that you can upload new media to folders in the media tree.
